Docs/source/manual/Implementation.rst

```diff
@@ -0,0 +1,197 @@
+calling : ::
+
+    auto ldata_p = getLevelDataPtr(lev,AmrOldTime);
+
+with either `AmrOldTime` or `AmrNewTime` on level `lev`. Additionnally, calling this function with
+`AmrHalfTime` with return a `LevelData` struct whose `state` is a linearly interpolated between the old and new
+states (but the other MultiFab in `LevelData` are empty !).
+It is also often useful to have access to a vector of a state component accross the entire AMR hierarchy. To do so, *PeleLMeX*
+provides a set of functions returning a vector of MultiFab `std::unique_ptr` aliased into the `LevelData`
+MultiFab on each level: ::
+
+    getStateVect(time);         # Return the entire state (ncomp: NVAR)
+    getVelocityVect(time);      # Return the velocity only (ncomp: AMREX_SPACEDIM)
+    getDensityVect(time);       # Return the mixture density (ncomp: 1)
+    getSpeciesVect(time);       # Return the species density (ncomp: NUM_SPECIES)
+    getRhoHVect(time);          # Return rhoH (ncomp: 1)
+    getTempVect(time);          # Return temperature (ncomp: 1)
+    getDivUVect(time);          # Return divergence constraint (ncomp: 1)
+    getDiffusivityVect(time);   # Return diffusivity (ncomp: NUM_SPECIES+2)
+    getViscosityVect(time);     # Return viscosity (ncomp: 1)
+
+where ``time`` can either be `AmrOldTime` or `AmrNewTime`.
+Also available at any point during the simulation is the `LevelDataReact` which contains the species
+chemical source terms. A single version of the container is avaible on each level and can be accessed
+using: ::
+
+    auto ldataR_p = getLevelDataReactPtr(lev);
+
+Within the time-advance function, the *PeleLMeX* algorithm calls for the computation of the advection,
+diffusion and reaction source terms iteratively using SDC. At each step, the results of other steps
+can be used as part of the numerical scheme (e.g. the explicit advection with a Godunov scheme uses
+the diffusion term). These temporary variables, only useful in the scope of the advance function, are
+assembled into two structs: ``AdvanceDiffData`` and ``AdvanceAdvData``. The former contains three
+MultiFabs for the separate diffusion term evaluations described in :numref:`LMeX_Algo`: :math:`D^n`,
+:math:`D^{n+1,k}` and :math:`D^{n+1,k+1}`, as well as additional containers for the :math:`\overline{W}`
+and Soret contributions. The later encapsulate the face-centered MAC velocities :math:`U_{ADV}`, the
+advection term :math:`A_{n+1/2,(k+1)}`, the pressure correction :math:`\chi` and a forcing container
+used in the RHS of advection/diffusion/reaction solves. In contrast with the `LevelData`, these two containers
+are freed at the end of the advance function, and are passed around in the functions called in `advance()`.
+
+Parallelism
+-----------
+
+*PeleLMeX* inherits the MPI+X approach from the AMReX library, where X can be any of CUDA, HIP or SYCL
+(Open-MP is currently unavailable) for heterogeneous architectures. The reader is referred to
```

Comment:
Rephrase, OpenMP is limited to nonreacting and explicit chemistry integration but nto supported for most reacting calculations.